sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala

```diff
@@ -109,13 +109,29 @@ object Encoders {
    */
   def DATE: Encoder[java.sql.Date] = ExpressionEncoder()
 
+  /**
+   * Creates an encoder that serializes instances of the `java.time.LocalDate` class
+   * to the internal representation of Catalyst's DateType.
```

Comment:
`DATE` and `TIMESTAMP` encoders' comments mention the nullability. Shall we also mention it here?

Reply:
Looking how the serializer for `java.time.LocalDate` is invoked at https://github.com/apache/spark/blob/34f606678a90e860711a5f9f9618cf00788c9eb0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SerializerBuildHelper.scala#L89 , the result is non-nullable `DateType`. I can mention that in the comment.

Also if I am right, comments for some other types are incorrect.

Reply:
No, I am wrong the result is nullable date:
```scala
spark.range(1).map { _ => localDate }.printSchema()
```
```
root
 |-- value: date (nullable = true)
```
I will update comments accordantly.